Data mining in health and wellness offers numerous impactful applications. It transforms raw nutrition and food tracking data into actionable insights, helping users make informed decisions that support their overall health and wellness goals.

In this practice activity, you will be working with a preliminary review of a nutrition and food tracking dataset. The main objective of this exercise is for you to understand the dataset structure and get familiar with the meaning of each feature of the nutritional data.

**Instructions:**

1. Download the nutrition dataset  and load it to .
    
    [here](https://drive.google.com/file/d/1sLoEkd84nwHpJK0yOhBhcsNzQyV_fd_X/view?usp=drive_link)
    
    [Jupyter Notebooks](https://jupyter.org/)
    
2. Inspect the first few rows and columns using head() to get a sense of the data.
3. Review the dataset’s features and identify potential target variables.
4. Based on the initial findings, answer the following questions:
    1. How many entries does the dataset have?
    2. What are the column names, and what kind of data do they contain?
    3. Do you notice any missing or unusual values?
5. Create a table that lists each feature, describes its meaning, and identifies its type (e.g., numerical, categorical).
6. Based on the results, answer the following questions:
    1. Which columns are numerical, and which are categorical?
    2. Are there any features that look redundant or unnecessary for analysis?

In [ ]:
# 1. Import Libraries
import pandas as pd

# 2. Load the Dataset
df = pd.read_csv('/nutrition_data.csv')

# Verify the dataset load
print(df.head())


        food_item  calories  protein  carbs  fats  meal_time
0           Apple      95.0      0.5   25.0   0.3  Breakfast
1          Banana     105.0      1.3   27.0   0.4      Snack
2  Chicken Breast     165.0     31.0    0.0   3.6      Lunch
3           Steak     679.0     62.0    0.0  48.0     Dinner
4           Salad     150.0      2.0   15.0   7.0      Lunch


Reviewing the dataset's feature.


In [ ]:
# 3. Dataset Structure
num_rows = df.shape[0]
num_cols = df.shape[1]

print(f"The dataset has {num_rows} rows and {num_cols} columns.")

print("\nDataset Info:")
df.info()

The dataset has 105 rows and 6 columns.

Dataset Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 105 entries, 0 to 104
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   food_item  105 non-null    object 
 1   calories   103 non-null    float64
 2   protein    104 non-null    float64
 3   carbs      104 non-null    float64
 4   fats       105 non-null    float64
 5   meal_time  105 non-null    object 
dtypes: float64(4), object(2)
memory usage: 5.0+ KB


The given dataset has 105 rows and 6 columns namely 'food_item', 'calories', 'protein', 'carbs', 'fats', and 'meal_time'

Checking any missing and unusual values in the given dataset.

In [ ]:
# 4. Missing Values
missing_values = df.isnull().sum()
print("\nMissing Values in the Dataset:")
print(missing_values)


Missing Values in the Dataset:
food_item    0
calories     2
protein      1
carbs        1
fats         0
meal_time    0
dtype: int64


In [ ]:
# 5. Feature Review Table
features = {
    "Feature": ["food_item", "calories", "protein", "carbs", "fats", "meal_time"],
    "Description": [
        "Name of the food item",
        "Calories in the food item",
        "Protein content (grams)",
        "Carbohydrate content (grams)",
        "Fat content (grams)",
        "Meal timing (e.g., breakfast, lunch, etc.)"
    ],
    "Type": ["Categorical", "Numerical", "Numerical", "Numerical", "Numerical", "Categorical"]
}

feature_review_df = pd.DataFrame(features)
print("\nFeature Review Table:")
print(feature_review_df)

# 6. Analyze Feature Types
numerical_columns = df.select_dtypes(include=['int64', 'float64']).columns.tolist()
categorical_columns = df.select_dtypes(include=['object']).columns.tolist()

print(f"\nNumerical Columns: {numerical_columns}")
print(f"Categorical Columns: {categorical_columns}")


Feature Review Table:
     Feature                                 Description         Type
0  food_item                       Name of the food item  Categorical
1   calories                   Calories in the food item    Numerical
2    protein                     Protein content (grams)    Numerical
3      carbs                Carbohydrate content (grams)    Numerical
4       fats                         Fat content (grams)    Numerical
5  meal_time  Meal timing (e.g., breakfast, lunch, etc.)  Categorical

Numerical Columns: ['calories', 'protein', 'carbs', 'fats']
Categorical Columns: ['food_item', 'meal_time']
